### PR TITLE
CASMPET-5846

### PR DESCRIPTION
**Pull request by mitchty**
# Description

Add initial docs for kubernetes encryption setup and usage. As well as add the script used to setup the encryption after install or upgrade.

As this script was largely unit tested for another repo bring in similar infrastructure for shellspec to keep the unit test framework in place for future updates or bugfixes.

This was tested largely on redbull where these examples come from, though I did scrub dates in example output.  I have validated on other systems before rebuild but there isn't much difference.

# Checklist Before Merging

- [x ] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [x ] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [x ] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

File: operations/encryption/Overview.md
Comment (mitchty) [inline operations/encryption/Overview.md:160]: Gah how did spellcheck not find this.

**Review by kimjensen-hpe — APPROVED** [review]

File: install/README.md
Comment (mharding-hpe) [inline install/README.md:210]:
```suggestion
As an optional post installation task, encryption of Kubernetes secrets may be enabled. This enables
at rest encryption of data in the `etcd` database used by Kubernetes.
```

File: install/README.md
Comment (mharding-hpe) [inline install/README.md:212]:
```suggestion
See [Kubernetes Encryption](../operations/encryption/Overview.md).
```

File: operations/encryption/Overview.md
Comment (mharding-hpe) [inline operations/encryption/Overview.md:7]:
```suggestion
By default, encryption is not enabled and must be enabled after install.
```

**Comment by mharding-hpe** [comment]
Adding "do not merge" label because I want to review this before we merge it (and have had bad luck with things being merged when I have only commented on this in my feedback but not added the label). This does not imply anything about my feelings as to the quality of the content here -- it just means that I don't know if I will have time to review this before the first time it will be considered for merging.

I will remove the label once I've finished my reviews and issues identified (if any) are addressed.

**Comment by mitchty** [comment]
Ok thanks for the quick review @mharding-hpe the biggest questionable bit here might be the shellspec unit test library but it is a separate commit so I can easily remove it. But I really want to have more unit tests of our shell in general and wanted to lead with examples. It was a part of a commit I didn't get in in time for the node images but I think in general our shell could use a few more unit tests.

File: scripts/operations/node_management/encryption.sh
Comment (mharding-hpe) [inline scripts/operations/node_management/encryption.sh:61]: I think I must be missing something here. I don't see this function ever being called in this script, and it seems odd to make a function which just returns 1. It feels like either:
1. This script sources some other shell file, and that file calls this function
2. Some other script sources THIS script, and that file calls this function.

But I don't see this script sourcing anything else. I didn't check to see if the other scripts in this PR sourced this file, but I do see that they never call the validate function that I could see. 

What am I missing?

File: operations/encryption/Overview.md
Comment (mharding-hpe) [inline operations/encryption/Overview.md:1]: If this is specifically for Kubernetes encryption, would it make more sense for this to be located in the `operations/kubernetes/encryption/` folder? And usually for the main file on a topic, we would name it README.md (and create a symlink for index.md pointing to it).

Alternatively, if we want this in the general encryption folder because we anticipate grouping it with other, non-k8s encryption topics, then our convention is to have the file name match its main header. So in this case, it would be Kubernetes_Encryption.md.

**Review by mharding-hpe — CHANGES_REQUESTED** [review]

**Comment by mharding-hpe** [comment]
> Ok thanks for the quick review @mharding-hpe the biggest questionable bit here might be the shellspec unit test library but it is a separate commit so I can easily remove it. But I really want to have more unit tests of our shell in general and wanted to lead with examples. It was a part of a commit I didn't get in in time for the node images but I think in general our shell could use a few more unit tests.

You will never hear me argue against more tests, believe me. :)

File: operations/encryption/Overview.md
Comment (mitchty) [reply operations/encryption/Overview.md:1]: I can change it to whatever makes sense. I'll swap it out for the latter. I was mostly following some other conventions I had seen.

File: scripts/operations/node_management/encryption.sh
Comment (mitchty) [reply scripts/operations/node_management/encryption.sh:61]: Probably leftovers from a red/green refactor, I'll nuke it.

**Comment by mitchty** [comment]
Ok I swapped the files into operations/kubernetes/encryption and scripts/operations/kubernetes respectively. Sat on it overnight but this seems a better layout with your suggestions. I don't think this will ever expand out beyond just kubernetes encryption in etcd really.

File: operations/kubernetes/encryption/README.md
Comment (rgross727) [inline operations/kubernetes/encryption/README.md:12]: [Enabling encryption](#enabling-encrypion)
spelling on encryption

Comment (rgross727) [reply operations/kubernetes/encryption/README.md]: Fixed now.

**Review by mharding-hpe — COMMENTED** [review]
My previous concerns appear to be resolved

**Review by mharding-hpe — APPROVED** [review]
I haven't had a chance to fully re-review all of the changes since the last time I reviewed it, but I did confirm that my original concerns appear to be addressed.

**Review by rustydb — APPROVED** [review]

**Pull request merged**
